iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineImpl.java

```diff
@@ -54,6 +215,11 @@
         this.g = g;
         this.b = b;
         this.ucteXnodeCode = ucteXnodeCode;
+        if (generationAdder != null) {
+            generation = new GenerationImpl(generationAdder, this);
+        } else {
+            generation = null;
+        }
```

Comment:
That's strange: the constructor should have a Generation parameter instead of mixing data and adder?

Reply:
The issue is `GenerationImpl` cannot be created if the `DanglingLineImpl` object has not being created... I am a bit bothered by it too but I did not have any better idea.

Reply:
Okay, I changed a bit the implementation in my last commit and put the same mecanism that with T3W's leg: a generation without a set `danglingline` is created at first and `danglingLine` is set when the object is created. It is a bit cleaner.